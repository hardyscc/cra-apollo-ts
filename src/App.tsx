import * as React from 'react'
import { Query } from 'react-apollo'
import { GetStocks as QUERY } from './queries'
import { GetStocksQuery } from './__generated__/types'

class StocksQuery extends Query<GetStocksQuery> {}

export const App = () => (
  <StocksQuery query={QUERY}>
    {({ loading, error, data }) => {
      if (loading) return <div>Loading...</div>
      if (error) return <div>Error :(</div>

      return (
        <div>
          Stocks :
          <ul>
            {data &&
              data.stocks.map(stock => (
                <li key={stock.code}>
                  {stock.code} {stock.name}
                </li>
              ))}
          </ul>
        </div>
      )
    }}
  </StocksQuery>
)
