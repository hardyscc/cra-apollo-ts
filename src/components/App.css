.menu-box {
  line-height: 64px;
}

.content-box {
  background-color: #fff;
  padding: 24px;
  min-height: 280px;
}
