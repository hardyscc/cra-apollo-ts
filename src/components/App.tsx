import * as React from 'react'
import { Query } from 'react-apollo'
import { gql } from 'apollo-boost'
import { GetStocksQuery } from '../__generated__/types'
import { Button } from 'antd'

class StocksQuery extends Query<GetStocksQuery> {}

export const STOCKS_QUERY = gql`
  query GetStocks {
    stocks(type: EQTY, first: 10, skip: 0) {
      code
      name
    }
  }
`

export const App = () => (
  <StocksQuery query={STOCKS_QUERY}>
    {({ loading, error, data }) => {
      if (loading) return <div>Loading...</div>
      if (error) return <div>Error :(</div>

      return (
        <div>
          <h4>Stocks :</h4>
          <ul>
            {data &&
              data.stocks.map(stock => (
                <li key={stock.name}>
                  {stock.code} {stock.name}
                </li>
              ))}
          </ul>
          <Button type="primary">Test</Button>
        </div>
      )
    }}
  </StocksQuery>
)
