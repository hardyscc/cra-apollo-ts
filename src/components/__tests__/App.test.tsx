import * as React from 'react'
import { render, wait } from 'react-testing-library'
import { MockedProvider } from 'react-apollo/test-utils'
import { App, STOCKS_QUERY } from '../App'

const stocks = [
  { code: '0001.HK', name: '長和' },
  { code: '0002.HK', name: '中電控股' }
]
const request = { query: STOCKS_QUERY }
const result = { data: { stocks } }
const error = new Error('something go wrong')

describe('App', () => {
  it('render', async () => {
    const { container, getByText } = render(
      <MockedProvider mocks={[{ request, result }]}>
        <App />
      </MockedProvider>
    )
    await wait(() => getByText('Stocks'))
    expect(container).toMatchSnapshot()
  })

  it('error', async () => {
    const { container, getByText } = render(
      <MockedProvider mocks={[{ request, error }]}>
        <App />
      </MockedProvider>
    )
    await wait(() => getByText('Error'))
    expect(container).toMatchSnapshot()
  })
})
