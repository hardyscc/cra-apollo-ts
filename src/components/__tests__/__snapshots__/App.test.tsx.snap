// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App error 1`] = `
<div>
  <div>
    Error :(
  </div>
</div>
`;

exports[`App render 1`] = `
<div>
  <div>
    <h4>
      Stocks :
    </h4>
    <ul>
      <li>
        0001.HK
         
        長和
      </li>
      <li>
        0002.HK
         
        中電控股
      </li>
    </ul>
  </div>
</div>
`;
